src/Microsoft.AspNetCore.DataProtection/DataProtectionProvider.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Cryptography.Cng;
+using Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption;
+using Microsoft.AspNetCore.DataProtection.Cng;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.AspNetCore.DataProtection.Repositories;
+using Microsoft.AspNetCore.DataProtection.XmlEncryption;
+using Microsoft.Extensions.Logging;
+using Microsoft.Win32;
+
+namespace Microsoft.AspNetCore.DataProtection
+{
+    public abstract class DataProtectionProvider : IDataProtectionProvider
```

Comment:
Yes. But this PR is still in design. I am planning to do all the documentation/test/renames in a later update.